src/Cms/Auth.php

```diff
@@ -308,28 +308,20 @@ public function impersonate(string|null $who = null)
 		// clear the status cache
 		$this->status = null;
 
-		switch ($who) {
-			case null:
-				return $this->impersonate = null;
-			case 'kirby':
-				return $this->impersonate = new User([
-					'email' => 'kirby@getkirby.com',
-					'id'    => 'kirby',
-					'role'  => 'admin',
-				]);
-			case 'nobody':
-				return $this->impersonate = new User([
-					'email' => 'nobody@getkirby.com',
-					'id'    => 'nobody',
-					'role'  => 'nobody',
-				]);
-			default:
-				if ($user = $this->kirby->users()->find($who)) {
-					return $this->impersonate = $user;
-				}
-
-				throw new NotFoundException('The user "' . $who . '" cannot be found');
-		}
+		return $this->impersonate = match ($who) {
+			null     => null,
+			'kirby'  => new User([
+				'email' => 'kirby@getkirby.com',
+				'id'    => 'kirby',
+				'role'  => 'admin',
+			]),
+			'nobody' => new User([
+				'email' => 'nobody@getkirby.com',
+				'id'    => 'nobody',
+				'role'  => 'nobody',
+			]),
+			default  => $this->kirby->users()->find($who) ?? throw new NotFoundException('The user "' . $who . '" cannot be found')
+		};
```

Comment:
PHPMD uses the PDepend library, which comes with its own PHP parser. I assume the false-positive error comes from a spin-off of this issue: https://github.com/pdepend/pdepend/pull/544

We could try this alternatively:

```suggestion
		$this->impersonate = match ($who) {
			null     => null,
			'kirby'  => new User([
				'email' => 'kirby@getkirby.com',
				'id'    => 'kirby',
				'role'  => 'admin',
			]),
			'nobody' => new User([
				'email' => 'nobody@getkirby.com',
				'id'    => 'nobody',
				'role'  => 'nobody',
			]),
			default  => $this->kirby->users()->find($who)
		};
		
		return $this->impersonate ?? throw new NotFoundException('The user "' . $who . '" cannot be found');
```

Reply:
Wait, this won't work because of the other cases. Maybe it's already enough to wrap the return value of the default case with `(...)`.